crates/project/src/project.rs

```diff
@@ -3201,6 +3201,51 @@ impl Project {
         let disk_based_diagnostics_progress_token =
             adapter.disk_based_diagnostics_progress_token.clone();
 
+        // Log LSP messages under experimental/serverStatus and display pop-up for errors
+        language_server
+            .on_notification::<ServerStatus, _>({
+                let this = this.clone();
+                let name = name.to_string();
+                move |params, mut cx| {
+                    let this = this.clone();
+                    let name = name.to_string();
+                    match params.health {
+                        ServerHealthStatus::Ok => log::info!("Health: Ok"),
+                        ServerHealthStatus::Warning => log::warn!(
+                            "Health: Warning.\nMessage: {}",
```

Comment:
I do not think we should use those `Health: Warning.` parts at all, as `warn!` seems already doing that for us?
What is missing here, is which language server emitted that. We have the `name` and `server_id`, let's mention those and remove `\n` from the logs.

In the end, we'll might get something like 
`log::LEVEL!("Language server {name} (id {server_id}) status update: {message}");`

I also do not think we should bother and print anything if the message is `None` or `message.trim().is_empty()`.

Same applies to all other places, including the `ServerHealthStatus::Other` match arm that has no message printed at all, which is weird — so I'm wondering if some common logging function should be used, that will match over the status and do the job?
What do you think?

Reply:
Applied most of the suggestions you've made. Regarding the common logging function, I think that could be a possibility, but I'm just not too sure how we might want to handle the `ServerHealthStatus::Other` case. 
In my code, I logged `Unknown server health` and included the status and message. Let me know what you think! Thank you!